plugins/aja/aja-card-manager.cpp

```diff
@@ -0,0 +1,618 @@
+{
+	return mCardIndex;
+}
+
+std::string CardEntry::GetCardID() const
+{
+	return mCardID;
+}
+
+std::string CardEntry::GetDisplayName() const
+{
+	if (mCard) {
+		std::ostringstream oss;
+		oss << mCard->GetIndexNumber() << " - "
+		    << mCard->GetModelName();
+		const std::string &serial = GetSerial();
+		if (!serial.empty())
+			oss << " (" << serial << ")";
+		return oss.str();
+	}
+
+	// very bad if we get here...
+	return "???";
+}
+
+std::string CardEntry::GetSerial() const
+{
+	std::string serial;
+	if (mCard)
+		mCard->GetSerialNumberString(serial);
+	return serial;
+}
+
+NTV2DeviceID CardEntry::GetDeviceID() const
+{
+	NTV2DeviceID id = DEVICE_ID_NOTFOUND;
+	if (mCard)
+		id = mCard->GetDeviceID();
+	return id;
+}
+
+bool CardEntry::ChannelReady(NTV2Channel chan, const std::string &owner) const
+{
+	const std::lock_guard<std::mutex> lock(mMutex);
+
+	bool found = false;
+
+	for (const auto &pwn : mChannelPwnz) {
+		if (pwn.second & (1 << static_cast<int32_t>(chan))) {
+			return pwn.first == owner;
+		}
+	}
+
+	return !found;
+}
+
+bool CardEntry::AcquireChannel(NTV2Channel chan, NTV2Mode mode,
+			       const std::string &owner)
+{
+	bool acquired = false;
+
+	if (ChannelReady(chan, owner)) {
+		const std::lock_guard<std::mutex> lock(mMutex);
+
+		if (mChannelPwnz.find(owner) != mChannelPwnz.end()) {
+			if (mChannelPwnz[owner] &
+			    (1 << static_cast<int32_t>(chan))) {
+				acquired = true;
+			} else {
+				mChannelPwnz[owner] |=
+					(1 << static_cast<int32_t>(chan));
+				acquired = true;
+			}
+		} else {
+			mChannelPwnz[owner] |=
+				(1 << static_cast<int32_t>(chan));
+			acquired = true;
+		}
+
+		// Acquire interrupt handles
+		if (acquired && mCard) {
+			if (mode == NTV2_MODE_CAPTURE) {
+				mCard->EnableInputInterrupt(chan);
+				mCard->SubscribeInputVerticalEvent(chan);
+			} else if (mode == NTV2_MODE_DISPLAY) {
+				mCard->EnableOutputInterrupt(chan);
+				mCard->SubscribeOutputVerticalEvent(chan);
+			}
+		}
+	}
+
+	return acquired;
+}
+
+bool CardEntry::ReleaseChannel(NTV2Channel chan, NTV2Mode mode,
+			       const std::string &owner)
+{
+	const std::lock_guard<std::mutex> lock(mMutex);
+
+	for (const auto &pwn : mChannelPwnz) {
+		if (pwn.first == owner) {
+			if (mChannelPwnz[owner] &
+			    (1 << static_cast<int32_t>(chan))) {
+				mChannelPwnz[owner] ^=
+					(1 << static_cast<int32_t>(chan));
+
+				// Release interrupt handles
+				if (mCard) {
+					if (mode == NTV2_MODE_CAPTURE) {
+						mCard->DisableInputInterrupt(
+							chan);
+						mCard->UnsubscribeInputVerticalEvent(
+							chan);
+					} else if (mode == NTV2_MODE_DISPLAY) {
+						mCard->DisableOutputInterrupt(
+							chan);
+						mCard->UnsubscribeOutputVerticalEvent(
+							chan);
+					}
+				}
+
+				return true;
+			}
+		}
+	}
+
+	return false;
+}
+
+bool CardEntry::InputSelectionReady(IOSelection io, NTV2DeviceID id,
+				    const std::string &owner) const
+{
+	// uncomment if we add device-specific logic
```

Comment:
That was a typo. The comment should rather indicate that the `UNUSED_PARAMETER(id)` is a placeholder for device-specific configuration for the capture plugin, following the structure of the other functions where device-specific configuration was implemented for the output plugin (`AcquireOutputSelection` and `ReleaseOutputSelection`).

Reply:
I think it would be apparent to most people who might modify these functions that if they added code to use the parameter in question (`id`) that they should remove the call to the `UNUSED_PARAMETER` macro.